drones/image_processing/yolo.py

```diff
@@ -0,0 +1,138 @@
+        self.stride = int(self.model.stride.max())  # model stride
+        self.conf_tres = float(self.config["CONFIDENCE_THRESHOLD"])
+        self.img_size = int(self.config["IMG_SIZE"])
+        self.iou_thres = float(self.config["IOU_THRESHOLD"])
+
+    def detect(self, img0: np.ndarray) -> list:
+        """Detect object on image using provided weights. Objects are detected by YOLO neural network
+
+        Parameters:
+        ----------
+            img0: np.ndarray
+                image on which objects detection will be proceeded
+        Returns:
+        ----------
+            It returns list of all detected objects on photo. Every element consist 5 values. First one is recognized
+            class name. Others are central position of object(x,y) and size of object(width, height).
+            All values are normalized in yolo norm.
+            To have position and size on original photo you have to multiply this results by original photo size.
+        """
+        # Load model
+
+        img_size = check_img_size(self.img_size, s=self.stride)  # check img_size
+        img = img0.copy()
+
+        img = letterbox(img, img_size, self.stride)[0]
+
+        # Convert
+        img = img[:, :, ::-1].transpose(2, 0, 1)  # BGR to RGB, to 3x416x416
+        img = np.ascontiguousarray(img)
+
+        # Get names and colors
+        names = self.model.module.names if hasattr(self.model, "module") else self.model.names
+
+        device = select_device(self.device_type)
+
+        # Run inference
+        if device.type != "cpu":
+            self.model(
+                torch.zeros(1, 3, img_size, img_size).to(device).type_as(next(self.model.parameters()))
+            )  # run once
+        t0 = time.time()
+        result = []
+        image = torch.from_numpy(img).to(device)
+        image = image.float()  # uint8 to fp16/32
+        image /= 255.0  # 0 - 255 to 0.0 - 1.0
+        if image.ndimension() == 3:
+            image = image.unsqueeze(0)
+
+        # Inference
+        pred = self.model(image)[0]
+
+        # Apply NMS
+        pred = non_max_suppression(pred, self.conf_tres, self.iou_thres, classes=self.classes)
+
+        # Process detections
+        for i, det in enumerate(pred):  # detections per image
+            gn = torch.tensor(img0.shape)[[1, 0, 1, 0]]  # normalization gain whwh
+            if len(det):
+                # Rescale boxes from img_size to im0 size
+                det[:, :4] = scale_coords(image.shape[2:], det[:, :4], img0.shape).round()
+
+                # Write results
+                for *xyxy, conf, cls in reversed(det):
+                    xywh = (xyxy2xywh(torch.tensor(xyxy).view(1, 4)) / gn).view(-1).tolist()  # normalized xywh
+                    line = (names[int(cls)], *xywh)  # label format
+                    result.append(line)
+
+        logging.getLogger(__name__).info(f"Processed image, processing time: ({time.time() - t0:.3f}s)")
```

Comment:
You can add debug log messages, containing positions of all detected objects (either here, or in `detect_object_yolo` below).

Reply:
Still log messages missing